exercicios/para-casa/entregas/julia-mariah/contaBancaria.test.js

```diff
@@ -0,0 +1,96 @@
+const {contaBancaria} = require("../julia-mariah/contaBancaria.js")
+const nomeBanco = new contaBancaria();
+
+nomeBanco.nomeTitular = 'julia';
+nomeBanco.saldo = 500;
+nomeBanco.limiteConta = 1000;
+
+describe("mostrar os dados da conta bancária", () => {
+    test("deve validar o nome", () => {
+        expect(nomeBanco.validaNome (nomeBanco.nomeTitular)).toEqual("julia");
+    })
+
+    test("deve mostrar o saldo", () => {
+        expect(nomeBanco.validaSaldo (nomeBanco.saldo)).toEqual(500);
+    })
+
+    test("deve mostrar o limite da conta", () => {
+        expect(nomeBanco.validaLimiteconta (nomeBanco.limiteConta)).toEqual(1000);
+    })
+    
+    test("fazer um saque" , () => {
+        expect(nomeBanco.fazerSaque (500)).toEqual(0);
+         
+    })
+
+    test("fazer um depósito" , () => {
+        expect(nomeBanco.fazerDeposito (200)).toEqual(1700);
+    })
+
+    test("mostrar limite atualizado" ,() => { 
+        nomeBanco.fazerSaque (600)
+        const resultado = nomeBanco.limiteConta
+        expect(resultado).toEqual(900)
```

Comment:
O raciocínio na função foi super correto. 
Aqui o erro do teste ocorre somente porque: a função saque altera o limite para saldo + limite 
                     this.limiteConta = this.limiteConta + this.saldo - Linha 20 do arquivo contaBancaria.js

A função fazerDeposito não está atualizando o saldo da conta. 

Assim, uma vez que o JS lê o arquivo de cima para baixo e a última atualização do saldo é de 0, quando você faz o saque de 600 reais a função retira o valor do limite, já que não existe saldo, e retorna 400. 

O teste está correto, mas o que deve ser ajustado é a função fazerDeposito atualizar o saldo.

Reply:
Verdade, olha eu amei suas dicas. Tomara eu consiga ficar boa nisso !